### PR TITLE
DRMPRIME: keep plane coords of video render area private

### PR DESCRIPTION
**Pull request by HiassofT**
## Description
Store plane coordinates of render area in a separate, private member
instead of re-using m_destRect so that m_destRect is again guaranteed
to be within m_viewRect.

## Motivation and context
Fixing #20085, see also #20173 for some background discussion

## How has this been tested?
Tested on LibreELEC / RPi4. Verified that subtitles where shown when GUI limit was active.

## What is the effect on users?
Subtitles will be visible, even if the GUI limit setting is used (and active).

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/BaseRenderer.cpp [hunk 1]
```diff
@@ -190,10 +196,26 @@ void CBaseRenderer::CalcNormalRenderRect(float offsetX, float offsetY, float wid
   else if (verticalShift < -1.0f)
     posY += shiftRange * (verticalShift + 1.0f);
 
-  m_destRect.x1 = (float)MathUtils::round_int(static_cast<double>(posX + offsetX));
-  m_destRect.x2 = m_destRect.x1 + MathUtils::round_int(static_cast<double>(newWidth));
-  m_destRect.y1 = (float)MathUtils::round_int(static_cast<double>(posY + offsetY));
-  m_destRect.y2 = m_destRect.y1 + MathUtils::round_int(static_cast<double>(newHeight));
+  destRect.x1 = (float)MathUtils::round_int(static_cast<double>(posX + offsetX));
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/BaseRenderer.cpp:199]: While you are at it, mind changing the c-style casts to a static_cast ?

Comment (HiassofT) [reply xbmc/cores/VideoPlayer/VideoRenderers/BaseRenderer.cpp]: sure, just pushed an update

**Review by Kwiboo — APPROVED** [review]
Looks good

**Comment by enen92** [comment]
Jenkins build and merge

**Pull request merged**